src/components/messaging/conversation.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import {
+  ImageProps,
+  View,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+import {
+  Conversation as ConversationModel,
+  Message,
+} from '@src/core/model';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import { DoneAllIcon } from '@src/assets/icons';
+import { Text } from '@src/components/common';
+import { ConversationInterlocutor } from './';
+
+interface ComponentProps {
+  index?: number;
+  conversation: ConversationModel;
+  onConversation: (index: number) => void;
+}
+
+export type ConversationProps = & ThemedComponentProps & TouchableOpacityProps & ComponentProps;
+
+class ConversationComponent extends React.Component<ConversationProps> {
+
+  private onConversation = (): void => {
+    this.props.onConversation(this.props.index);
+  };
+
+  private getLastMessageText = (): string => {
+    const { conversation } = this.props;
+    const lastMessage: string = conversation.messages[conversation.messages.length - 1].text;
+
+    return lastMessage.length <= 37 ? lastMessage : `${lastMessage.substring(0, 37)}...`;
+  };
+
+  private getLastMessageDate = (): string => {
+    const { conversation } = this.props;
+
+    return conversation.messages[conversation.messages.length - 1].date;
+  };
+
+  private renderLastMessageIcon = (): React.ReactElement<ImageProps> | null => {
+    const { conversation, themedStyle } = this.props;
+    const lastMessage: Message = conversation.messages[conversation.messages.length - 1];
+    const lastMessageRead: boolean = lastMessage.read;
+    const lastMessageDelivered: boolean = lastMessage.delivered;
+
+    if (lastMessageRead) {
```

Comment:
how about using enums for message status which can provide icons?

Reply:
Done